cifs/parse_cifs.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cifs
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// ParseClientStats returns stats read from /proc/fs/cifs/Stats
+func ParseClientStats(r io.Reader) (*ClientStats, error) {
```

Comment:
Please refactor this into at least four or five smaller functions.  There is too much context to keep track of here, and any indentation beyond 3 or so levels becomes unreadable.

Reply:
sure absolutly!